pkg/controllers/allocation/binpacking/packable.go

```diff
@@ -164,16 +166,28 @@ func (p *Packable) validateArchitecture(schedule *scheduling.Schedule) error {
 	return nil
 }
 
-func (p *Packable) validateOperatingSystem(schedule *scheduling.Schedule) error {
-	if schedule.Requirements.OperatingSystems().Intersection(p.OperatingSystems()).Len() == 0 {
-		return fmt.Errorf("operating system %s is not in %v", p.OperatingSystems(), schedule.Requirements.OperatingSystems().List())
+func (p *Packable) validateZones(schedule *scheduling.Schedule) error {
+	packableZoneSet := sets.String{}
+	for _, offering := range p.Offerings() {
+		packableZoneSet.Insert(offering.Zone)
+	}
+	if schedule.Requirements.Zones().Intersection(packableZoneSet).Len() == 0 {
+		return fmt.Errorf("zones %v are not in %v", packableZoneSet, schedule.Requirements.Zones().List())
 	}
 	return nil
 }
 
-func (p *Packable) validateZones(schedule *scheduling.Schedule) error {
-	if schedule.Requirements.Zones().Intersection(p.Zones()).Len() == 0 {
-		return fmt.Errorf("zones %v are not in %v", p.Zones(), schedule.Requirements.Zones().List())
+func (p *Packable) validateCapacityTypes(schedule *scheduling.Schedule) error {
+	scheduledCapacityTypes := schedule.Requirements.Requirement(v1alpha1.CapacityTypeLabel)
+	if scheduledCapacityTypes.Len() == 0 {
+		return nil
+	}
```

Comment:
TBH I'm not entirely sure if this is correct, but the tests fail without it. Need to get more sleep and read more code to think through this.